File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int sum = 0;
        for (size_t num = 0; num < words.size(); ++num) {
            const string& word = words[num];
            sum += count_if(words.begin(), words.begin() + num, [&word](const string& w) {
                return word.substr(0, w.size()) == w && word.substr(word.size() - w.size()) == w;
            });
        }
        return sum;
    }
};

File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        result = 0
        for num, word in enumerate(words):
            result += sum(1 for w in words[:num] if word.startswith(w) and word.endswith(w))
        return result





